client/boot/locale.js

```diff
@@ -10,13 +10,25 @@ import { loadUserUndeployedTranslations } from 'calypso/lib/i18n-utils/switch-lo
 import { LOCALE_SET } from 'calypso/state/action-types';
 import { setLocale } from 'calypso/state/ui/language/actions';
 
-export function getLocaleFromPathname() {
+export function getLocaleFromPathname( options ) {
+	const { excludedPathSegments = [] } = options || {};
 	const pathname = window.location.pathname.replace( /\/$/, '' );
+
+	// The excludedPathSegments option can be used on paths where the last path segment is the resource slug.
+	// This option avoids the possible collision between the resource slug and the locale slug.
+	if ( excludedPathSegments.length ) {
+		const penultimatePathSegment = pathname.split( '/' ).at( -2 );
```

Comment:
Hmm, should we count from the beginning, not from the end?

I think it's intuitive if `excludedPathSegments` is the penultimate segment; e.g. what happens if in the future someone creates a flow at `/start/with-free-promotion/theme/ja`; it will hit this logic, which (most likely) is unintended.

Whereas, if we define it as the first segment, then we can control it -- we know that it is "the" `/theme/<slug>` (theme details) page, not anything else.

Reply:
I remember there are places that assume the locale from right to left, so I'm not sure whether it's possible to count from the beginning 🥲

Reply:
Yep, like `https://wordpress.com/zh-tw/theme/pendant` has the locale at the beginning 😅

Reply:
Ohh.... I see, then I guess we don't have much choice here 🥲